sycl/unittests/Extensions/CommandGraph/Update.cpp

```diff
@@ -399,3 +399,14 @@ TEST_F(WholeGraphUpdateTest, EmptyNode) {
   auto GraphExec = Graph.finalize(experimental::property::graph::updatable{});
   GraphExec.update(UpdateGraph);
 }
+
+TEST_F(CommandGraphTest, CheckFinalizeBehavior) {
+  // Check that both finalize with and without updatable property work as
+  // expected
+  auto Node = Graph.add(
+      [&](sycl::handler &cgh) { cgh.single_task<TestKernel<>>([]() {}); });
+
+  ASSERT_NO_THROW(Graph.finalize(experimental::property::graph::updatable{}));
+
+  ASSERT_NO_THROW(Graph.finalize());
+}
```

Comment:
I think it would make sense to do a `set_after_callback` for some of the new functions and make a counter to check that the functions are called the expected number of times. See sycl/unittests/context_device/DeviceRefCounter.cpp for example.

Reply:
Good idea thanks, I've added some counting of the calls for this test now.